docs/tutorial/batch-minting.md

```diff
@@ -0,0 +1,268 @@
+  struct Unminted {
+    address player;
+    string tokenURI;
+  }
+
+  // Keep record of unminted backlogs as a dynamic
+  // array of Unminted objects.
+  Unminted[] private backlogs;
+
+  // Declare the feed contract.
+  EACAggregatorProxy public gasfeed;
+
+  // Pass the EACAggregatorProxy contract address to the contructor
+  // in order to initialize the gasfeed contract.
+  constructor(address feed) ERC721("GameItem", "ITM") {
+    gasfeed = address
+  }
+
+  function awardItem(address player, string memory tokenURI)
+      public
+      returns (uint256)
+  {
+      uint256 newItemId = _tokenIds.current();
+
+      // Mint only if the gas price is satisfactory. Otherwise, we push
+      // to the backlogs array for to be minted later.
+      if (getLatestGasPrice() <= MAX_GAS_PRICE) {
+        _mint(player, newItemId);
+        _setTokenURI(newItemId, tokenURI);
+        _tokenIds.increment();
+
+        // Tide is high! Let's mint the backlogs while the gas price is low.
+        if (backlogs.length > 0) {
+          for (uint i=0; i<backlogs.length; i++) {
+            _mint(backlogs[i].player, backlogs[i].tokenURI);
+            _tokenIds.increment();
+            _setTokenURI(_tokenIds.current(), backlogs[i].tokenURI);
+          }
+        }
+
+      } else {
+        backlogs.push(Unminted(player, tokenURI));
+      }
+
+      return newItemId;
+  }
+
+  function getLatestGasPrice() publivc view returns (uint256)  {
+    var (_roundId, answer, _startedAt, _updatedAt, _answeredInRound) = gasfeed.latestRoundData();
+    return answer;
+  }
+}
+
+```
+
+To deploy the contract, don't forget to pass the Chainlink’s aggregator contract address into deploy function of the contract factory:
+
+```javascript
+
+const AGGREGATOR_ADDRESS = "0x169E633A2D1E6c10dD91238Ba11c4A708dfEF37C"
+let GameItemFactory = await ethers.getContractFactory("GameItem")
+let gameItem = await GameItemFactory.deploy(AGGREGATOR_ADDRESS)
+await gameItem.deployed()
+console.log(`GameItem deployed to: ${gameItem.address}`)
+
+```
+
+### ERC-721A: An Alternative
+
+[ERC-721A](https://www.erc721a.org/) is an alternative ERC-721 implementation by the Azuki NFT development team. It was designed primarily for batch-minting with very low gas fees compared to conventional ERC-721 and ERC-1155. According to the developers, the new algorithm enables minting multiple NFTs for essentially the same cost as minting a single NFT at the expense of more complexity by means of several optimization tactics, such as removal of duplicate storage from OpenZeppelin’s ERC721Enumerable and updating the owner’s balance and other data once per batch mint request instead of per minted NFT.
+
+## Conclusion
+
+There is a few ways to approach batch-upload and -minting hundreds or thousands of your NFTs and their assets on NFT.storage and Ethereum. However, the most recommended way is to employ `uploadDirectory` to upload files on NFT.storage and `ERC-1155` for batch-minting on Ethereum.
```

Comment:
```suggestion
There are multiple ways to approach batch-upload and -minting hundreds or thousands of your NFTs and their assets on NFT.storage and Ethereum. However, the most recommended way is to employ `uploadDirectory` or NFTUp to upload files on NFT.storage, and `ERC-1155` for batch-minting on Ethereum.
```

do you mean storeDirectory (not uploadDirectory)?

Reply:
@Yes that was a typo.